### DISCUSSION TITLE
Use of ujson

### DISCUSSION THREAD

**Discussion by HAK-CODE** [aws-powertools/powertools-lambda-python#1189] [Q&A] [answered]
Can powertools maintainers use [ujson](https://github.com/ultrajson/ultrajson) for encoding and decoding dicts, as it may be helpful in performance improvements.

**Comment by michaelbrewer** [comment]
You can swap out which json library you want to use, for the event, the dict was parsed before the lambda has been invoked.

There are examples of bring in your own serializer: https://awslabs.github.io/aws-lambda-powertools-python/latest/core/logger/#bring-your-own-json-serializer

**Comment by heitorlessa** [comment, marked as answer]
@HAK-CODE the reason we don't use `ujson` by default is that `ujson` it's not a fully compatible drop-in replacement. This could potentially break customers use cases we can't foresee. Instead, we prefer providing a mechanism to bring your own JSON serializer - for example, [orjson ](https://github.com/ijl/orjson) as it's even faster than `ujson`.

If we are not providing you a mechanism to do that for a given feature, please don't hesitate to create a [feature request](https://github.com/awslabs/aws-lambda-powertools-python/issues/new?assignees=&labels=feature-request%2Ctriage&template=feature_request.yml&title=Feature+request%3A+TITLE).
